### PR TITLE
Added support for Redis cluster

### PR DESCRIPTION
**Pull request by cadu-goncalves**
Created support for Redis cluster based on JedisCluster

### CODE REVIEW

File: src/main/scala/com/globo/bigdata/gedis/JedisFactory.scala [hunk 1]
```diff
@@ -18,6 +18,11 @@ class JedisFactory {
     new Jedis(s.host, s.port, s.connectionTimeout, s.socketTimeout)
   }
 
+  def jedis(s: ClusterSettings) : JedisCluster = {
+    import scala.collection.JavaConverters._
```
Comment (sbouchardet) [inline src/main/scala/com/globo/bigdata/gedis/JedisFactory.scala:46]: Hi =)
why did you import `scala.collection.JavaConverters._` inside the function?

Comment (cadu-goncalves) [reply src/main/scala/com/globo/bigdata/gedis/JedisFactory.scala:46]: Hi.
To use the `asJava` method on `s.clusterNodes.asJava`

**Review by dmvieira — CHANGES_REQUESTED** [review]
Hi @cadu-goncalves , please just fix tests

**Review by cadu-goncalves — COMMENTED** [review]
> Hi @cadu-goncalves , please just fix tests

Sure!

File: src/main/scala/com/globo/bigdata/gedis/JedisFactory.scala (hunk 1 above)
Comment (dmvieira) [reply src/main/scala/com/globo/bigdata/gedis/JedisFactory.scala:46]: LGTM

**Comment by cadu-goncalves** [comment]
Sorry about the delay, in fact the previous strategy using reflection was really difficult to adjust. I've just simplified the solution mapping the calls inside the adapter class itself.
**More verbose** than previous approach however **removing reflection provides more performance**

**Comment by dmvieira** [comment]
Good point! Great solution

**Pull request merged**